File: example-app/src/test/java/com/nickmcdowall/example/FishClient.java
package com.nickmcdowall.example;

import feign.RequestInterceptor;
import org.springframework.cloud.openfeign.FeignClient;
import org.springframework.context.annotation.Bean;
import org.springframework.web.bind.annotation.DeleteMapping;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.PathVariable;
import org.springframework.web.bind.annotation.PostMapping;

@FeignClient(name = "fishClient", url = "${server.url}", configuration = FishClient.FishClientConfig.class)
public interface FishClient {

    @PostMapping(value = "/fish")
    void post(NewFishRequest request);

    @GetMapping("/fish/{name}")
    String getFishWithName(@PathVariable String name);

    @DeleteMapping("/fish/{name}")
    void deleteByName(@PathVariable String name);

    @PostMapping(value = "/fish/{id}/{name}")
    void createFish(@PathVariable long id, @PathVariable String name);

    class FishClientConfig {
        @Bean
        public RequestInterceptor requestInterceptor() {
            return template -> {
                template.header("Source-Name", "User");
                template.header("Target-Name", "FishApp");
            };
        }
    }
}

File: example-app/src/test/java/io/lsdconsulting/intercceptors/example/FishClient.kt
package io.lsdconsulting.intercceptors.example

import feign.RequestInterceptor
import feign.RequestTemplate
import io.lsdconsulting.intercceptors.example.FishClient.FishClientConfig
import org.springframework.cloud.openfeign.FeignClient
import org.springframework.context.annotation.Bean
import org.springframework.web.bind.annotation.*

@FeignClient(name = "fishClient", url = "\${server.url}", configuration = [FishClientConfig::class])
interface FishClient {
    @PostMapping(value = ["/fish"])
    fun post(request: NewFishRequest?)

    @GetMapping("/fish/{name}")
    fun getFishWithName(@PathVariable name: String?): String?

    @GetMapping("/fish")
    fun getFishByName(@RequestParam("name") name: String?): List<String?>?

    @DeleteMapping("/fish/{name}")
    fun deleteByName(@PathVariable name: String?)

    @PostMapping(value = ["/fish/{id}/{name}"])
    fun createFish(@PathVariable id: Long, @PathVariable name: String?)
    class FishClientConfig {
        @Bean
        fun requestInterceptor(): RequestInterceptor {
            return RequestInterceptor { template: RequestTemplate ->
                template.header("Source-Name", "User")
                template.header("Target-Name", "FishApp")
            }
        }
    }
}
